### DISCUSSION TITLE
Postgres Bigint - PBKDF2 iterations minimum is 5000

### DISCUSSION THREAD

**Discussion by sheepnaniganz** [dani-garcia/vaultwarden#2905] [General]
Not sure where to post this, but when migrating to postgres from sqlite on first run it created the columns that were to be integers as bigint.

I had to manually change them all to integer in order for clients to work. 

I looked through the schema and it is listed as integer, so I am not sure why it created them as bigint. 

Just leaving this here for anyone who has the same issue as I could not find anything about this.

Also when loggin in it was giving a error of `PBKDF2 iterations minimum is 5000` which is what led me to the issue.

**Comment by RealOrangeOne** [comment]
5000 rounds of PBKDF2 is pretty weak. Bitwarden [uses](https://bitwarden.com/help/what-encryption-is-used/#pbkdf2) 100000 by default, which is much more reasonable, but even then I've seen applications use 300000.

Using less than 5000 is like using nothing at all, so having the database prevent you from doing it doesn't feel too bad.

**Reply by BlackDex** [reply]
Just as a note, Vaultwarden uses the same as bitwarden. The minimum message was because of his low database int i think
